### PR TITLE
[feature](nereids): implementation rules.

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (no)
2. Has unit tests been added: no
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/properties/SortSpec.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+
+package org.apache.doris.nereids.properties;
+
+/**
+ * Base class for Sort.
+ */
+public class SortSpec {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/properties/SortSpec.java:24]: why add this class？

**Comment by EmmyMiao87** [comment]
@924060929  is canceling the operator in the plan. Your rule conflicts with his logic. 
Do you need to wait for his incorporation before incorporating your pr?

**Comment by morrySnow** [comment]
> @924060929 is canceling the operator in the plan. Your rule conflicts with his logic. Do you need to wait for his incorporation before incorporating your pr?

merge operator with plan is a BIG change for Nereids. IMO, it is better to make Nereids could WORK on SSB benchmark first and then apply this change. Although this will be a bit more work, but there are two advantages. First, we could verify that merging operator and plan into one data structure is feasible easily. Second, After we have verified the Nereids using the SSB benchmark, we can focus on work unrelated to the main process that will not be affected by this change.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**